scripts/migrate_dimension.py

```
@@ -0,0 +1,100 @@
+        for i, memory in enumerate(all_memories):
+            if i % 10 == 0 or i == total - 1:
+                logger.info(f"Processing {i+1}/{total} (ID: {memory.id})")
+            
+            try:
+                # Re-embed content
+                new_embedding = await embedding_provider.embed(memory.content)
+                # Update in storage (convert memory.id to string to avoid UUID binding error)
+                success = await storage.update_memory(str(memory.id), {"embedding": new_embedding})
+                if not success:
+                    logger.warning(f"Failed to update memory {memory.id}")
+            except Exception as e:
```

Ruff (0.15.10)

[warning] Do not catch blind exception: `Exception`

(BLE001)